atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleNamespaceDeleterTests.java

```diff
@@ -0,0 +1,374 @@
+    }
+
+    private Set<String> listAllPhysicalTableNames() {
+        return sqliteOracleAdapter.listAllPhysicalTableNames();
+    }
+
+    private Set<TableReference> listAllTablesWithMetadata() {
+        return sqliteOracleAdapter.listAllTablesWithMetadata();
+    }
+
+    private Set<String> listAllTablesWithMapping() {
+        return sqliteOracleAdapter.listAllTablesWithMapping();
+    }
+
+    private NamespaceDeleter createDefaultNamespaceDeleter() {
+        return new OracleNamespaceDeleter(
+                createDefaultNamespaceDeleterParameters().build());
+    }
+
+    private ImmutableOracleDdlConfig.Builder getDefaultDdlConfig() {
+        return ImmutableOracleDdlConfig.builder()
+                .tablePrefix(TABLE_PREFIX)
+                .overflowTablePrefix(OVERFLOW_TABLE_PREFIX)
+                .overflowMigrationState(OverflowMigrationState.UNSTARTED)
+                .useTableMapping(true)
+                .metadataTable(SqliteOracleAdapter.METADATA_TABLE);
+    }
+
+    private ImmutableOracleNamespaceDeleterParameters.Builder createDefaultNamespaceDeleterParameters() {
+        return createNamespaceDeleterParameters(
+                getDefaultDdlConfig().build(),
+                new ConnectionSupplier(sqliteOracleAdapter.createSqlConnectionSupplier()));
+    }
+
+    private ImmutableOracleNamespaceDeleterParameters.Builder createNamespaceDeleterParameters(
+            ConnectionSupplier connectionSupplier) {
+        return createNamespaceDeleterParameters(getDefaultDdlConfig().build(), connectionSupplier);
+    }
+
+    private ImmutableOracleNamespaceDeleterParameters.Builder createNamespaceDeleterParameters(
+            OracleDdlConfig ddlConfig) {
+        return createNamespaceDeleterParameters(
+                ddlConfig, new ConnectionSupplier(sqliteOracleAdapter.createSqlConnectionSupplier()));
+    }
+
+    private ImmutableOracleNamespaceDeleterParameters.Builder createNamespaceDeleterParameters(
+            OracleDdlConfig ddlConfig, ConnectionSupplier connectionSupplier) {
+        OracleTableNameGetter tableNameGetter = OracleTableNameGetterImpl.createDefault(ddlConfig);
+        Function<TableReference, OracleDdlTable> ddlTableFactory = tableReference -> OracleDdlTable.create(
+                tableReference,
+                connectionSupplier,
+                ddlConfig,
+                tableNameGetter,
+                new TableValueStyleCache(),
+                executorService);
+        return OracleNamespaceDeleterParameters.builder()
+                .tablePrefix(ddlConfig.tablePrefix())
+                .overflowTablePrefix(ddlConfig.overflowTablePrefix())
+                .userId(TEST_USER)
+                .oracleDdlTableFactory(ddlTableFactory)
+                .connectionSupplier(connectionSupplier)
+                .tableNameGetter(tableNameGetter);
+    }
+
+    private class UnstableConnectionSupplier extends ConnectionSupplier {
```

Comment:
Would happily take an alternative way of doing this, but my goal is to throw in the middle of deleting some tables - I guess I could mock out the factory function and have that throw? But that got quite messy with extracting out the dependency tree such that we continue deleting after the fact